scalableminds/webknossos
# Allow to continue upload after browser restart for same user

### URL of deployed dev instance (used for testing):
- https://morerobustresumableupload.webknossos.xyz/

### Steps to test:
- Open the upload view on the dev instance: https://morerobustresumableupload.webknossos.xyz/datasets/upload?to=66b60af101000001003deac5
- Open the dev tools. In the debugger open file `dataset_upload_view.tsx` and set a breakpoint in line 425 (before `resumableUpload.upload();` is called upon the files added action.
- Keep the dev tools open, upload some small dataset (to have less traffic) 
  - Enter dataset name, allowed teams, target folder and drag & drop a dataset zip
- Hit the upload button. The upload should be reserved and then the debugger should kick in an interrupt starting the upload process. 
- Close your browser and do not skip the breakpoint.
- Reopen the browser and reopen the upload view.
- Open the dev tools to track the network traffic
- The view should now show the dataset name you just tried to upload but manually interrupted before starting the upload (stopping during the upload should also work. Click the continue upload
- The UI should be autofilled with the previous values and disabled.
- Drag & drop the same file as before and hit upload
- The upload should happily continue and succeed :)
- Now test the network traffic. There should be **get** requests to the `/datasets` route testing whether a chunk is already present. They should be returning 204 if no chunk was previously uploaded. If you tweaked the testing such that some chunks were already uploaded, some of these requests should answer 200 (signaling the chunk is present) (see below)
- Test the uploaded dataset. It should be viewable as usual. 




##
- Alternatively, for the debugger interruption, choose a larger dataset to upload and interrupt the upload in time.

### TODOs:
- [x] Test whether this works with multi-file upload (e.g. multiple tiffs at once)

### Issues:
- fixes [#7931](https://github.com/scalableminds/webknossos/issues/7931)

------
(Please delete unneeded items, merge only when none are left open)
